### PR TITLE
Instant Search: Improve browser compatibility + UX for input handling

### LINKED ISSUE #18553: Instant Search: InputEvent.inputType API is not supported in IE11
#### Steps to reproduce the issue

1. Setup a site with Jetpack Instant Search. 
2. In Internet Explorer 11, select a search input and start typing.

#### What I expected
The search modal should have appeared.

#### What happened instead
Nothing happens. Instead, the browser console throws an error saying `SCRIPT5007: Unable to get property 'includes' of undefined or null reference`. This is because IE11 does not support [inputType](https://caniuse.com/mdn-api_inputevent_inputtype) used [here](https://github.com/Automattic/jetpack/blob/0a12031ffa7ec6c873d366cbf443bf85edc21c23/projects/plugins/jetpack/modules/search/instant-search/components/search-app.jsx#L162).

### PR DESCRIPTION
**Pull request by jsnmoon**
Fixes #18553.

#### Changes proposed in this Pull Request:
* Ignore `event.inputType` in the event handler if not supported by browser.
* Immediately search and spawn the overlay when a user presses Enter on a search input.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
* Apply this change to your site with Jetpack Search enabled.
* Ensure that your `node_modules` are up-to-date -- use `yarn disclean` if necessary.
* Try searching your site in IE11 by typing into a search input. Ensure that the overlay appears as expected.

#### Proposed changelog entry for your changes:
* None.

### CODE REVIEW

File: projects/plugins/jetpack/modules/search/instant-search/components/search-app.jsx [hunk 1]
```diff
@@ -159,11 +159,24 @@ class SearchApp extends Component {
...
 	handleInput = debounce( event => {
 		// Reference: https://rawgit.com/w3c/input-events/v1/index.html#interface-InputEvent-Attributes
-		if ( event.inputType.includes( 'format' ) || event.target.value === '' ) {
+		// NOTE: inputType is not compatible with IE11, so we use optional chaining here. https://caniuse.com/mdn-api_inputevent_inputtype
+		if ( event.inputType?.includes( 'format' ) ) {
+			return;
+		}
+
+		// If user presses enter, propagate the query value and immediately show the results.
+		if ( event.key === 'Enter' ) {
```
Comment (bluefuton) [inline projects/plugins/jetpack/modules/search/instant-search/components/search-app.jsx:168]: This doesn't work for me in FF or Chrome. I don't see a `key` property on the InputEvent:

Screen Shot 2021-01-28 at 14 15 09

File: projects/plugins/jetpack/modules/search/instant-search/components/search-app.jsx [hunk 2]
```diff
@@ -159,11 +159,24 @@ class SearchApp extends Component {
 
 	handleInput = debounce( event => {
 		// Reference: https://rawgit.com/w3c/input-events/v1/index.html#interface-InputEvent-Attributes
-		if ( event.inputType.includes( 'format' ) || event.target.value === '' ) {
+		// NOTE: inputType is not compatible with IE11, so we use optional chaining here. https://caniuse.com/mdn-api_inputevent_inputtype
+		if ( event.inputType?.includes( 'format' ) ) {
```
Comment (bluefuton) [inline projects/plugins/jetpack/modules/search/instant-search/components/search-app.jsx:163]: 👍

File: projects/plugins/jetpack/modules/search/instant-search/components/search-app.jsx (hunk 1 above)
Comment (jsnmoon) [reply projects/plugins/jetpack/modules/search/instant-search/components/search-app.jsx:164]: Good catch. Fixed this by creating a separate keydown handler and moving the necessary code over.

**Review by bluefuton — APPROVED** [review]
🟢 Tests out well for me, including on IE11.

**Review by jeherve — APPROVED** [review]
This tests well for me, in Chrome and IE11.

**Comment by jeherve** [comment]
Cherry-picked to `jetpack/branch-9.4` in d4ba62d2e2390165752366f7f4d839268e1602b9

**Comment by jsnmoon** [comment]
Deployed in r220180-wpcom.

**Pull request merged**